httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/AbstractMultipartPartBuilder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.client5.http.entity.mime;
+
+/**
+ * Abstract Builder for individual {@link org.apache.hc.client5.http.entity.mime.MultipartPart}s.
+ *
+ * @since 5.0
+ */
+public abstract class AbstractMultipartPartBuilder<T extends MultipartPart, S extends AbstractMultipartPartBuilder<T, S>> {
```

Comment:
@adamretter I would not create an abstract class just to re-use two instance variables. Would you mind getting rid of it? 

All other things look good to me. If you could fix the failing test cases and I would merge the change-set. 

Some test coverage for `MultipartPartBuilder` would also be nice.